sqlglot/parser.py

```diff
@@ -2205,9 +2205,14 @@ def _parse_distributed_property(self) -> exp.DistributedByProperty:
 
     def _parse_duplicate(self) -> exp.DuplicateKeyProperty:
         self._match_text_seq("KEY")
-        expressions = self._parse_wrapped_csv(self._parse_id_var, optional=False)
+        expressions = self._parse_wrapped_id_vars()
         return self.expression(exp.DuplicateKeyProperty, expressions=expressions)
 
+    def _parse_unique_property(self) -> exp.UniqueKeyProperty:
+        self._match_text_seq("KEY")
+        expressions = self._parse_wrapped_id_vars()
+        return self.expression(exp.UniqueKeyProperty, expressions=expressions)
```

Comment:
It could be worth consolidating this and `_parse_duplicate` into a single definition that gets a `Type[Expression]` and instantiates it according to this (common) logic.

Reply:
I don't get what do you want me to do here.

Reply:
Something like:

```python
def _parse_composite_key_property(self, expr_type: t.Type[E]) -> E:
    self._match_text_seq("KEY")
    expressions = self._parse_wrapped_id_vars()
    return self.expression(expr_type, expressions=expressions
```

So we can use it for both `_parse_unique_property` and `_parse_duplicate` in the base parser.

Reply:
Fixed.

Reply:
Thanks